sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -485,6 +486,74 @@ private[sql] object DataSourceStrategy extends Strategy with Logging {
   }
 
   /**
+   * Convert add predicate such as C1 + C2 + C3 to string
+   */
+  private[this] def getAddString (predicate: Expression): String = {
+    predicate match {
+      case expressions.Add(left, right) =>
+      {
+        val leftString = left match {
+          case a: Attribute => a.name
+          case add: Add => getAddString (add)
+          case _ => None
+        }
+        val rightString = right match {
+          case a: Attribute => a.name
+          case add: Add => getAddString (add)
+          case _ => None
+        }
+        leftString + " + " + rightString
+      }
+    }
+  }
+
+  /**
+   * Tries to translate a Catalyst [[Expression]] into data source [[Filter]].
```

Comment:
@viirya 
Thank you very much for your comments.  I changed the code based on your suggestion.  Since the new code is quite different from the original, I will close this PR and submit a new one.